From 679deaaf05ef18550c7262716effad45983aadab Mon Sep 17 00:00:00 2001
From: Luiz Augusto von Dentz <luiz.von.dentz@intel.com>
Date: Mon, 11 Aug 2014 17:58:37 +0300
Subject: [PATCH 6/9] Revert "gdbus: Don't include just added interfaces in
 GetManagedObjects"

This actually creates a problem since the code now does flush any pending
message on the queue any signal will be send right away before the
actual reply to GetManagedObjects which will not contain those interfaces.

(cherry picked from commit d98648a08cdf9d6d5ea9f32c1c63daa4baef877a)
Signed-off-by: Andreas Oberritter <obi@opendreambox.org>
---
 gdbus/object.c | 8 +-------
 1 file changed, 1 insertion(+), 7 deletions(-)

diff --git a/gdbus/object.c b/gdbus/object.c
index 13cf9a9..4d5a64c 100644
--- a/gdbus/object.c
+++ b/gdbus/object.c
@@ -1088,7 +1088,6 @@ static const GDBusMethodTable introspect_methods[] = {
 static void append_interfaces(struct generic_data *data, DBusMessageIter *iter)
 {
 	DBusMessageIter array;
-	GSList *l;
 
 	dbus_message_iter_open_container(iter, DBUS_TYPE_ARRAY,
 				DBUS_DICT_ENTRY_BEGIN_CHAR_AS_STRING
@@ -1100,12 +1099,7 @@ static void append_interfaces(struct generic_data *data, DBusMessageIter *iter)
 				DBUS_DICT_ENTRY_END_CHAR_AS_STRING
 				DBUS_DICT_ENTRY_END_CHAR_AS_STRING, &array);
 
-	for (l = data->interfaces; l != NULL; l = l->next) {
-		if (g_slist_find(data->added, l->data))
-			continue;
-
-		append_interface(l->data, &array);
-	}
+	g_slist_foreach(data->interfaces, append_interface, &array);
 
 	dbus_message_iter_close_container(iter, &array);
 }
-- 
1.9.1

